builtin/mainmenu/tab_multiplayer.lua

```diff
@@ -199,16 +233,122 @@ local function main_button_handler(tabview, fields, name, tabdata)
 		return true
 	end
 
+	if fields.btn_mp_search or fields.key_enter_field == "te_search" then
+		tabdata.fav_selected = 1
+		local input = fields.te_search:lower()
+		tabdata.search_for = fields.te_search
+		
+		if input == "" then 
+			menudata.search_result = nil
+			return true 
+		end
+		
+		if #menudata.favorites < 2 then 
+			return true 
+		end
+		
+		menudata.search_result = {}
+		
+		-- setup the filter and keyword list
+		local filters = {}
+		local keywords = {}
+		for word in input:gmatch("%S+") do
+			local index = word:find(":")
+			if index then
+				local filter = word:sub(1,index-1)
+				local setting = core.is_yes(word:sub(index+1,-1))
+
+				if filter == "pvp" then
+					filters.pvp = setting
+				elseif filter == "creative" then
+					filters.creative = setting
+				elseif filter == "version" then
+					if setting ~= "" then
```

Comment:
setting is a boolean, so will never equal "". I'm guessing you meant to preserve `word:sub(index + 1, -1)`